b2share/modules/deposit/api.py

```diff
@@ -246,3 +300,72 @@ def create_file_pids(record_metadata):
                 raise e
             else:
                 current_app.logger.warning(e)
+
+
+def find_version_master_and_previous_record(version_of):
+    try:
+        child_pid = RecordUUIDProvider.get(version_of).pid
+    except PIDDoesNotExistError as e:
+        raise RecordNotFoundVersioningError() from e
+
+    version_master = retrieve_version_master(child_pid)
+    if not version_master:
+        # the old record is NOT versioned yet
+        parent_pid = VersionMasterPidProvider.create().pid
+        parent_pid.redirect(child_pid)
+        version_master = PIDVersioning(parent=parent_pid)
+        version_master.insert_child(child=child_pid, index=-1)
+
+    prev_pid = version_master.last_child
```

Comment:
It doesn't seem to me that this code use the "draft" feature provided by invenio-pidrelations.